static/app/actionCreators/metrics.tsx

```diff
@@ -0,0 +1,75 @@
+import {Client} from 'sentry/api';
+import {getInterval} from 'sentry/components/charts/utils';
+import {DateString, MetricsApiResponse, Organization} from 'sentry/types';
+import {getPeriod} from 'sentry/utils/getPeriod';
+
+export type DoMetricsRequestOptions = {
+  field: string[];
+  orgSlug: Organization['slug'];
+  cursor?: string;
+  end?: DateString;
+  environment?: Readonly<string[]>;
+  groupBy?: string[];
+  includeAllArgs?: boolean;
+  interval?: string;
+  limit?: number;
+  orderBy?: string;
+  project?: Readonly<number[]>;
+  query?: string;
+  start?: DateString;
+  statsPeriod?: string | null;
+  statsPeriodEnd?: string | null;
+  statsPeriodStart?: string | null;
```

Comment:
Would we expect nulls for those values anytime or can they just be optional?

Reply:
Changed `statsPeriodStart` and `statsPeriodEnd` to be just optional 👍 
`statsPeriod` needs to support both because of how periods are handled in the rest of the app.